homeassistant/helpers/template.py

```diff
@@ -2178,45 +2198,62 @@ def _render_with_context(
         return template.render(**kwargs)
 
 
-class LoggingUndefined(jinja2.Undefined):
+def make_logging_undefined(
+    strict: bool | None, log_fn: Callable | None
+) -> type[jinja2.Undefined]:
     """Log on undefined variables."""
 
-    def _log_message(self) -> None:
-        template, action = template_cv.get() or ("", "rendering or compiling")
-        _LOGGER.warning(
-            "Template variable warning: %s when %s '%s'",
-            self._undefined_message,
-            action,
-            template,
-        )
+    if strict:
+        return jinja2.StrictUndefined
 
-    def _fail_with_undefined_error(self, *args, **kwargs):
-        try:
-            return super()._fail_with_undefined_error(*args, **kwargs)
-        except self._undefined_exception as ex:
+    if log_fn is None:
+        log_error = _LOGGER.warning
+        log_warning = _LOGGER.warning
+    else:
+        # Mypy does not agree log_fn is not None here
+        log_error = log_warning = lambda msg, *args: log_fn(msg % args)  # type: ignore[misc]
```

Comment:
Should we pass `warning` or `error` as an `level` arg ?

Reply:
That's a good idea, implemented 👍